Tasks/Grunt/grunttask.ts

```diff
@@ -41,11 +47,85 @@ gt.pathArg(gruntFile);
+	} else {
+		var istanbul = tl.createToolRunner(tl.which('./node_modules/istanbul/lib/cli.js', true));
+	}
+
+	istanbul.argString('cover --report cobertura');
+	istanbul.arg('./node_modules/mocha/bin/_mocha');
+	istanbul.arg(testSrc);
+	istanbul.argString('-- --ui bdd -t 5000');
+
+	var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
+
+	var summaryFile = path.join(buildFolder, 'coverage/cobertura-coverage.xml');
+}
+
+var publishJUnitResults = tl.getBoolInput('publishJUnitResults');
+var testResultsFiles = tl.getInput('testResultsFiles', publishJUnitResults);
+
+npm.exec().then(function () {
+	gt.exec()
+		.then(function (code) {
+			if (enableCoverage) {
+				istanbul.exec().then(function (code) {
+					publishTestResults(publishJUnitResults, testResultsFiles);
+					publishCodeCoverage(summaryFile);
+					tl.setResult(tl.TaskResult.Succeeded, tl.loc('GruntReturnCode', code));
+				})
+					.fail(function (err) {
+						publishTestResults(publishJUnitResults, testResultsFiles);
+						tl.debug('taskRunner fail');
+						tl.setResult(tl.TaskResult.Failed, tl.loc('IstanbulFailed', err.message));
+					});
+			} else {
+				publishTestResults(publishJUnitResults, testResultsFiles);
+				tl.setResult(tl.TaskResult.Succeeded, tl.loc('GruntReturnCode', code));
+			}
+
+		})
+		.fail(function (err) {
+			publishTestResults(publishJUnitResults, testResultsFiles);
+			tl.debug('taskRunner fail');
+			tl.setResult(tl.TaskResult.Failed, tl.loc('GruntFailed', err.message));
+		})
+}).fail(function (err) {
 	tl.debug('taskRunner fail');
-	tl.setResult(tl.TaskResult.Failed, tl.loc('GruntFailed', err.message));
-})
\ No newline at end of file
+	tl.setResult(tl.TaskResult.Failed, tl.loc('NpmFailed', err.message));
+})
+
+function publishTestResults(publishJUnitResults, testResultsFiles: string) {
+    if (publishJUnitResults) {
+        //check for pattern in testResultsFiles
+        if (testResultsFiles.indexOf('*') >= 0 || testResultsFiles.indexOf('?') >= 0) {
+            tl.debug('Pattern found in testResultsFiles parameter');
+            var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
+            var allFiles = tl.find(buildFolder);
+            var matchingTestResultsFiles = tl.match(allFiles, testResultsFiles, { matchBase: true });
+        }
+        else {
+            tl.debug('No pattern found in testResultsFiles parameter');
+            var matchingTestResultsFiles = [testResultsFiles];
+        }
+
+        if (!matchingTestResultsFiles) {
+            tl.warning('No test result files matching ' + testResultsFiles + ' were found, so publishing JUnit test results is being skipped.');
+            return 0;
+        }
+
+        var tp = new tl.TestPublisher("JUnit");
+        tp.publish(matchingTestResultsFiles, true, "", "", "", true);
+    }
+}
+
+function publishCodeCoverage(summaryFile) {
+	var ccPublisher = new tl.CodeCoveragePublisher();
+	ccPublisher.publish('cobertura', summaryFile, "", "");
+}
```

Comment:
i think by default you should generate HTML reports too and attach it as report directory